doc/source/drivers/raster/vrt_processed_dataset.rst

```diff
@@ -0,0 +1,261 @@
+
+
+The following optional arguments may be specified:
+
+- ``src_nodata``: Override the input nodata value coming from the previous step (or the input dataset for the first step).
+
+- ``dst_nodata``: Set the output nodata value.
+
+- ``replacement_nodata``: Value to substitute to a valid computed value that would be equal to dst_nodata.
+
+- ``dst_intended_datatype``: Intended datatype of output (which might be different than the working data type). Used to infer an appropriate value for replacement_nodata when it is not specified.
+
+- ``min``: Clamp minimum value, applied before writing the output value.
+
+- ``max``: Clamp maximum value, applied before writing the output value.
+
+
+Trimming algorithm
+------------------
+
+Apply local thresholding of saturation, with a special processing of the R,G,B bands compared to other bands.
+
+The pseudo algorithm used for each pixel is:
+
+.. code-block::
+
+    // Extract local saturation value from trimming image
+    localMaxRGB = value from TrimmingImage
+    reducedRGB = min ( (1-top_margin)*top_rgb/localMaxRGB ; 1)
+
+    // RGB bands specific process
+    RGB[] = get red, green, blue components of input buffer
+    maxRGB = max(RGB[])
+    toneMaxRGB = min ( toneCeil/maxRGB ; 1)
+    toneBand[] = min ( toneCeil/RGB[] ; 1)
+
+    output_value_RGB[] = min ( reducedRGB*RGB[]*toneBand[] / toneMaxRGB ; topRGB)
+
+    // Other bands processing (NIR, ...): only apply RGB reduction factor
+    Trimmed(OtherBands[]) = reducedRGB * OtherBands[]
+
+
+The following required arguments must be specified:
+
+- ``trimming_dataset_filename``: Filename of the trimming dataset. It must have one single band. It must have a georeferencing consistent of the input dataset, but may have a different resolution.
+
+- ``top_rgb``: Maximum saturating RGB output value.
+
+- ``tone_ceil``: Maximum threshold beyond which we give up saturation.
+
+- ``top_margin``: Margin to allow for dynamics in brighest areas (between 0 and 1, should be close to 0)
+
+
+The following optional arguments may be specified:
+
+- ``relative_filenames``: Whether the trimming dataset filename is relative to the VRT. Allowed values are ``true`` and ``false``. Defaults to ``false``
+
+- ``red_band``: Index (one-based) of the red band. Defaults to 1.
+
+- ``green_band``: Index (one-based) of the green band. Defaults to 1.
+
+- ``blue_band``: Index (one-based) of the blue band. Defaults to 1.
+
+- ``nodata``: Override the input nodata value coming from the previous step (or the input dataset for the first step).
+
+- ``trimming_nodata``: Override the nodata value coming from the trimming dataset.
+
+
+LUT
+---
+
+Apply a look-up table (band per band), typically to get from UInt16 to Byte data types.
+
+The following required argument must be specified:
+
+- ``lut_{band}``: List of the form ``[src value 1]:[dest value 1],[src value 2]:[dest value 2],....``. {band} must be replaced by 1 to the number of bands.
+
+
```

Comment:
Note that interpolation will be used if pixel values are not found in the LUT. (maybe make this opt-out?)

Reply:
I'm not sure to have understood your comment. What happens here is the same as the LUT option of a VRTComplexSource, that is if the input value is lower than the lowest src value 1, the output value is dest value 1. And if the input value is greater than the maximum src value N, the output value is dest value N. So this clamps to those bounds

Reply:
I mean that that with LUT `1:10,3:30`, I would have expected a pixel value of `2` to either remain `2` or end up as nodata (these are the two choices in https://docs.qgis.org/3.34/en/docs/user_manual/processing_algs/qgis/rasteranalysis.html#reclassify-by-table)

So it would be good to clarify that the algorithm performs linear interpolation if a pixel value is not found in the LUT.

Reply:
> So it would be good to clarify that the algorithm performs linear interpolation if a pixel value is not found in the LUT.

ok, got it. Done per 07b6ab5ce6b15ed4fe94072762266a3acc553898